### ISSUE TITLE
Form error using * symbol

### ISSUE THREAD

**Issue by rlambert27** [wet-boew/wet-boew#6501]
Hello,

I noticed today that the Form validation example (http://wet-boew.github.io/v4.0-ci/demos/formvalid/formvalid-en.html) is using the * symbol to identify required fields. The field also has the "required" text.

I believe the * symbol is being inserted using :before on those labels that have the required class.

So this invokes two questions:

1) Why are we using two methods to identify the required fields?
2) Using the * symbol seems to prevent conformance to H90 sufficient technique since the technique mentions that "If a symbol is used, the user is advised of its meaning before the first use.". In the example form there is no mention of what the * symbol means. The test for this technique seems pretty clear "For each indicator of required status that is not provided in text, check that the meaning of the indicator is explained before the form control that uses it.".

Thoughts?

Roch Lambert

**Comment by LaurentGoderre** [comment]
I don't think it fails since the required status is also conveyed in text

**Comment by thomasgohard** [comment]
1) I don't see a problem with communicating the same information using two methods in this case. The asterisk as an indicator of a required field is fairly well established. The plain text "required" provides a backup for people who don't know what the asterisk stands for.

2) I think the current implementation fails to meet "If a symbol is used, the user is advised of its meaning before the first use." But I don't think that it fails "For each indicator of required status that is not provided in text, check that the meaning of the indicator is explained before the form control that uses it." as the text indicator defines the non-text indicator.

**Comment by LaurentGoderre** [comment]
Even then, the asterisk is added via CSS so it's not read to screen reader and the relationship is clear visually

**Comment by rlambert27** [comment]
Well just to play devil's advocate, the "each indicator of required status that is not provided in text" means the * symbol that is used. The "required" text can't be used to indicate the meaning of the * symbol because not only is it in the same form control it's also after the symbol and not before.

I have no issues using both items to indicate required status although I think the * symbol has become redundant with such a clear meaning of the word required in the label. 

We could leave both of them in the label but before the form add an explanation indicating "The * symbol in the following form indicates a required field".

**Comment by LaurentGoderre** [comment]
Btw, you are quoting a technique, not a common failure so not meeting this technique does not mean that we don't meet the success criterion

**Comment by thomasgohard** [comment]
According to the article [CSS generated content and screen readers](http://www.456bereastreet.com/archive/201205/css_generated_content_and_screen_readers/), some screen readers do read CSS generated content.

I just tested using the test page linked to from the article with NVDA and Firefox: The CSS generated content is being read by NVDA.

**Comment by thomasgohard** [comment]
Testing on the form validation page it doesn't read the asterisk, though.

**Comment by CalvinRodo** [comment]
Could the * cause an issue with someone with a learning disability, a11y isn't just for Visually disabled people.

**Comment by CalvinRodo** [comment]
Also it seems a bit redundant to have both

**Comment by thomasgohard** [comment]
The asterisk should improve scannability for sighted users. I'm not sure what the effect would be for people with learning disabilities.

We would need to do some usability testing with people with different types of disabilities to identify how they understand and interact with those elements.

**Comment by biljanap** [comment]
Anything new on this subject? I am in the process of implementing required fields and form validation for one of my applications and I would like to know if I should continue to rely on existing form validation example where both methods are used. Thanks.

**Comment by pjackson28** [comment]
I can't remember the issue number but this was discussed at length. In essence the two approaches serve different audiences. Originally we were using just the "required" at the end of the label as that makes it programmatically available to all users (in particular screen reader users and text browser users). We then added the red asterisk at the start of the label (through CSS) because it was requested for visual users to aid with scannability and usability (since the red asterisk at the start of a label is a common convention, even if when used on its own it can lead to accessibility and usability issues).

So the answer is until there is compelling evidence to demonstrate that the dual approach is doing more harm than good or that one approach on its own meets all needs, then both methods should continue to be used. Also from the learning disability standpoint, if a user has issues with understanding what the red asterisk means, they do have the fallback of the actual meaning being spelled out in text at the end of the label.

**Comment by hsrudnicki** [comment]
Can this issue be closed or otherwise?

**Comment by pjackson28** [comment]
Closing the issue since there is no WCAG failure (since the information is both programmatically available and provided explicitly in plain language text) or any evidence that the approach is causing more harm that good. Also note that the <code>required</code> and <code>aria-required</code> attributes are being used as well meaning we are conveying that the field is required through four different mechanisms.
